lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -46,17 +46,31 @@ static int getMMAVersionSafe(int computeCapability, tt::DotOp op) {
 SmallVector<unsigned, 2>
 warpsPerTileV2(tt::DotOp dotOp, const ArrayRef<int64_t> shape, int numWarps) {
   auto filter = [&dotOp](Operation *op) {
-    return op->getParentRegion() == dotOp->getParentRegion();
+    return op->getParentRegion() == dotOp->getParentRegion() &&
+           !isa<tt::TransOp>(op);
   };
   auto slices = multiRootGetSlice(dotOp, {filter});
-  for (Operation *op : slices)
+  bool hasChainedDot = false;
+  for (Operation *op : slices) {
     if (isa<tt::DotOp>(op) && (op != dotOp)) {
-      if (shape[0] >= shape[1]) {
-        return {(unsigned)numWarps, 1};
-      } else {
-        return {1, (unsigned)numWarps};
+      auto chainedDot = cast<tt::DotOp>(op);
+      if (auto mmaEncoding = chainedDot.getResult()
+                                 .getType()
+                                 .cast<RankedTensorType>()
+                                 .getEncoding()
+                                 .dyn_cast<MmaEncodingAttr>()) {
+        return ttg::getWarpsPerCTA(mmaEncoding);
       }
+      hasChainedDot = true;
```

Comment:
How about add break to exit from for loop early?

Reply:
that would not work as there may be another dot in the slice that already has an mma encoding in which case I want this one to match.

Reply:
Ah, I see. Thanks!